### PR TITLE
Move build integration tests out of :buildSrc project

### PR DESCRIPTION
**Pull request by alpar-t**
This way we won't spend time with  `:buildSrc:check` for all invocations of Gradle, and it will be less disruptive if integration tests fail - at least being able to import to Idea.

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -152,6 +156,18 @@ if (project != rootProject) {
...
 
+  // tests can't  be run with randomized test runner
+  // it's fine as we run them as part of :buildSrc
+  test.enabled = false
+  task integTest(type: Test) {
+    exclude "**/*Tests.class"
+    include "**/*IT.class"
+    testClassesDirs = sourceSets.test.output.classesDirs
+    classpath = sourceSets.test.runtimeClasspath
+    inputs.dir(file("src/testKit"))
```
Comment (ywelsch) [inline buildSrc/build.gradle]: why is this necessary now?

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -152,6 +156,18 @@ if (project != rootProject) {
...
   thirdPartyAudit.enabled = false
 
+  // tests can't  be run with randomized test runner
+  // it's fine as we run them as part of :buildSrc
+  test.enabled = false
+  task integTest(type: Test) {
+    exclude "**/*Tests.class"
+    include "**/*IT.class"
+    testClassesDirs = sourceSets.test.output.classesDirs
+    classpath = sourceSets.test.runtimeClasspath
```
Comment (ywelsch) [inline buildSrc/build.gradle]: why not project.test.classpath?

File: buildSrc/build.gradle [hunk 3]
```diff
@@ -152,6 +156,18 @@ if (project != rootProject) {
   jarHell.enabled = false
   thirdPartyAudit.enabled = false
 
+  // tests can't  be run with randomized test runner
+  // it's fine as we run them as part of :buildSrc
+  test.enabled = false
+  task integTest(type: Test) {
+    exclude "**/*Tests.class"
+    include "**/*IT.class"
+    testClassesDirs = sourceSets.test.output.classesDirs
```
Comment (ywelsch) [inline buildSrc/build.gradle]: why not project.test.testClassesDirs?

**Review by ywelsch — CHANGES_REQUESTED** [review]
I've left some questions.

File: buildSrc/build.gradle (hunk 1 above)
Comment (atorok) [reply buildSrc/build.gradle]: This doesn't really belong here, it got pulled in as I tried to separate this PR out, and forgot to remove it.  The  integration tests use these sample Gradle projects. What this line does is makes sure tests are not considered up to date if only this directory changes.

File: buildSrc/build.gradle (hunk 3 above)
Comment (atorok) [reply buildSrc/build.gradle]: I might be wrong on this, but I think the source sets are the record of truth on this. 
Considering that we add a Gradle Test task and not a randomized one, I think it makes more sense than getting it from the randomzied testing task that we don't even use.

File: buildSrc/build.gradle (hunk 1 above)
Comment (ywelsch) [reply buildSrc/build.gradle]: ok

File: buildSrc/build.gradle (hunk 3 above)
Comment (ywelsch) [reply buildSrc/build.gradle]: ok

**Review by ywelsch — APPROVED** [review]
LGTM

**Review by rjernst — APPROVED** [review]
LGTM too
